docs/CI-CD.md

```diff
@@ -0,0 +1,93 @@
+
+## Publish
+
+This is the only action that uses both x86 and ARM64 runners. It gets triggered on either updates
+to the `master` branch or creation of release tags. Publish produces a set of tagged artifacts and
+pushes them into EVE's OCI Registry containers. Logic responsible for figuring out correct tags
+for these artifacts resides in EVE's top-level Makefile and can be summarized as:
+
+* If Publish gets triggered on a release tag (release tags are the ones with `x.y.z` names) that
+  compares as grater than all the other release tags currently present in a repo that release tag
+  is considered to be the latest. All artifacts originating from the Publish triggered on the latest
+  tag get a matching `x.y.z` OCI registry tags AND get an additional symbolic tag `latest`. Since
+  the artifacts are typically retrieved with a `docker pull lfedge/eve-XXX` this policy corresponds
+  well with the expectation that such a pull is expected to retrieve `lfedge/eve-XXX:latest` even
+  though it is a mere alias for `lfedge/eve-XXX:x.y.z`. Another crucial distinction of a Publish
+  action triggered on the latest tag is that it gets the benefit of a full (from scratch) build
+  as opposed to using pre-cached artifacts. This is done to ensure that EVE builds can be rebuilt
+  in the future in a self-contained manner but it does, of course, increase the build time dramatically.
+* If Publish gets triggered on a release tag that is NOT considered to be the `latest` (for example when
+  building .z patch releases for older x.y.(z-1) releases) the OCI registry artifacts only get a matching
+  `x.y.z` tag and no additional aliases are generated. The build is also typically much faster since it
+  uses pre-cached artifacts in OCI registry if the corresponding source package in the EVE tree didn't
+  change. Note, however, that building a patch release for the previous latest tag becomes a new latest
+  tag and thus subject to the latest tag treatment (e.g. `x.y.z` being latest means `x.y.(z+1)` will now
+  compare as greater than any other tag since it is greater than `x.y.z` itself).
+* If Publish gets triggered on a `master` branch all EVE packages in an OCI registry get two tags assigned:
+    1. symbolic tag `snapshot` (to facilitate `docker pull lfedge/eve-pillar:snapshot` type of usage)
+    2. source-derived unique tag
+  Source-derived unique tag is expected to be a SHA-like ID that changes only when any part of the
+  source tree corresponding to the EVE packages changes and thus allows for lazy (CAS-like) caching
+  of build updates. For all packages with actual source code this is linuxkit generate git tree sha
+  (e.g. `lfedge/eve-pillar:b8c0b1813a7819ee9eefdc77f1883da5613e7017`) for rootfs and its package this
+  is a synthetic version ID generated by the makefile of the form `0.0.0-master-<git sha>`.
+* If Publish gets triggered on any other branch (although this is rare and is not supported by default
```

Comment:
Then maybe we should just disable it and remove it? It is confusing.

Reply:
I kind of debated it -- but we actually do use it from time to time.